core/services/blockhashstore/feeder.go

```diff
@@ -246,6 +243,9 @@ func (f *Feeder) runTrusted(
 			errs = multierr.Append(errs, errors.Wrap(err, "checking if stored"))
 			return errs
 		}
+		for _, block := range blocksToStore {
```

Comment:
@vreff does this part look good to you? thought it would be more efficient to update `f.stored` after `StoredTrusted()`

Reply:
I think we need to get rid of the local `f.stored` mapping entirely, since now we could be storing blocks before finality depth and not try again after a re-org. It seems redundant since we always check if it's stored on-chain anyway. Or, maybe instead of the `f.stored` mapping we simply add it to an inflight queue that lasts for N seconds, rather than waiting for the entire lookback window. I know we have a cache component that you can just reuse: (https://github.com/smartcontractkit/chainlink/blob/feature/VRF-567-mercury-registry/core/services/ocr2/plugins/ocr2vrf/coordinator/ocr_cache.go#L29).

As it is, this logic doesn't really work anymore:
```
if f.lastRunBlock != 0 {
		// Prune stored, anything older than fromBlock can be discarded
		for block := f.lastRunBlock - uint64(f.lookbackBlocks); block < fromBlock; block++ {
			if _, ok := f.stored[block]; ok {
				delete(f.stored, block)
				f.lggr.Debugw("Pruned block from stored cache",
					"block", block, "latestBlock", latestBlock)
			}
		}
	}
```
Because we would:
1. Store,
2. See a re-org, and do nothing.
3. Free up the stored block after it has exited the lookback window, at which point it won't be storeable.

Reply:
I believe once a tx is accepted by TXM, it's guaranteed to make it on-chain even after re-org. Similarly, VRF listener marks a request as "processed" or "consumed" after fulfillment is written to TXM. If we remove `f.stored` completely, wouldn't the feeder spend too much gas?

Reply:
For the Trusted BHS I would just add the requests to an in-flight queue after submitting them, or seeing if we can't communicate with the TXM to track them

Reply:
yeah good catch for trusted BHS. I ended up making a new data structure called `storedTrusted`, which is a mapping from block number to blockhash. Upon re-org, trusted bhs should now can use previous blockhash to decide whether `storeTrusted` should be retried